### PR TITLE
new module panos_query_rules

### PR DESCRIPTION
**Pull request by ivanbojer**
##### SUMMARY
- Security policies allow you to enforce rules and take action, and can be as general or specific as needed. The policy rules are compared against the incoming traffic in sequence, and because the first rule that matches the traffic is applied, the more specific rules must precede the more general ones.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
panos_query_rule

##### ANSIBLE VERSION
```
ansible 2.4.2.0
  config file = None
  configured module search path = [u'/Users/ibojer/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/ibojer/.virtualenvs/ansible/lib/python2.7/site-packages/ansible
  executable location = /Users/ibojer/.virtualenvs/ansible/bin/ansible
  python version = 2.7.13 (default, Feb  6 2017, 11:22:10) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

### CODE REVIEW

**Comment by ivanbojer** [comment]
ready_for_review

File: lib/ansible/modules/network/panos/panos_query_rules.py [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+#  limitations under the License.
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: panos_query_rules
+short_description: search for security rules matching specific criteria
```
Comment (gundalow) [inline lib/ansible/modules/network/panos/panos_query_rules.py]: Please look at other `short_descriptions` for examples. Capital letter, mention product name, etc.

**Review by gundalow — CHANGES_REQUESTED** [review]
All network modules since 2.4 must include unit test. Have a look at `test/unit/modules/network/`

Fairly close to the 2.5 cut off.

File: lib/ansible/modules/network/panos/panos_query_rules.py (hunk 1 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_query_rules.py]: Updates short description.

**Comment by ivanbojer** [comment]
ready_for_review

**Comment by ivanbojer** [comment]
ready_for_review

**Comment by gundalow** [comment]
What's the plan for tests?

**Comment by ivanbojer** [comment]
Given that we just learned about this requirement there is no time to do them before the new release gets cut. We will add tests to all modules in the next iteration.

**Comment by ivanbojer** [comment]
ready_for_review

**Comment by gundalow** [comment]
@ivanbojer I'll merge this into `devel` fro 2.5.0, please as a priority work on adding some tests that are runnable via Shippable. Shout out in `#ansible-network` and/or `#ansible-devel` if you need help

**Review by gundalow — APPROVED** [review]

**Pull request merged**